### ISSUE TITLE
META: Make changes for extension registry MVP

### ISSUE THREAD

**Issue by jpmckinney** [open-contracting/ocds-extensions#59]
~~Some of these changes will be blocked by [extension_registry issues](https://github.com/open-contracting/extension_registry/issues/85).~~

Make updates to all extensions' extension.json:

* ~~Add "1.0" to `compatibility` array~~
  * Moved to #66
* ~~Add new "schemas" field, update extension-schema.json, decide whether required or optional~~
  * Moved to https://github.com/open-contracting/standard-maintenance-scripts/pull/99 and https://github.com/open-contracting/standard_extension_template/pull/34
* ~~Add new "slug" field, update extension-schema.json, decide whether required or optional~~
  * Slug (Id) is presently stored in the extension registry only
* [ ] Remove "documentationUrl" once extensions documentation website is available from extension.json, standard_extension_template, extension-schema.json, [Reviewing community extensions](https://docs.google.com/document/d/1bRhVVkuTPXw6acE2opKD-Yj80y0lvAsfAqDjG632nNY/edit), as this should be different for each version of each extension (adding overhead for extension authors), and as the website will centralize documentation at stable, predictable URLs
* ~~Use "version" in all extensions~~
  * Moved to https://github.com/open-contracting/standard-maintenance-scripts/pull/98

Other tasks:

* ~~Create releases for community extensions (not following OCDS version numbers)~~
    * ~~Pin the dependencies of `ocds_transactions_relatedMilestone_extension`~~
* Stop using GitHub URLs for extensions, where possible
    * [ ] extension.json: documentationUrl
    * [ ] ocds_transactions_relatedMilestone_extension: README.md
    * [ ] standard: extensions/index.md (via extension registry)
    * [x] ~~entry.json: documentationUrl~~
      * no longer exists
* ~~CoVE's behavior should be to error if it tries to fetch a schema listed in `schemas` and fails, but report no error if there are no `schemas` specified~~
  * Moved to https://github.com/OpenDataServices/cove/issues/1024

**Comment by jpmckinney** [comment]
@timgdavies The above requires making releases of community extensions. We can pick any version nomenclature we want, e.g. "1.0.0-beta". Are all community extensions "beta", or are some "alpha", and others neither?

**Comment by jpmckinney** [comment]
Although it may be preferable to have releases of community extensions, it's not mandatory for the registry and extensions documentation website, so crossing it out.

**Comment by jpmckinney** [comment]
@duncandewhurst Are there any extensions that you know are incompatible with 1.0? I think such compatibility is undeclared only by accident.

**Comment by duncandewhurst** [comment]
On a technical level:

* [Contract Suppliers](https://github.com/open-contracting/ocds_contract_suppliers_extension/blob/master/README.md), [Budget Breakdown](https://github.com/open-contracting/ocds_budget_breakdown_extension/blob/master/README.md), [Multiple buyers - contract level](https://github.com/open-contracting/ocds_multiple_buyers_extension), [Finance](https://github.com/open-contracting/ocds_finance_extension), [Requirements](https://github.com/open-contracting/ocds_requirements_extension/blob/master/README.md), [Shareholders](https://github.com/open-contracting/ocds-shareholders-extension) and [Contract Signatories](https://github.com/open-contracting/ocds_contract_signatories_extension/blob/master/README.md),   are incompatible with 1.0 because they reference `organizationReference`, which doesn't exist in 1.0
* [PPP Extension](https://github.com/open-contracting/public-private-partnerships/blob/master/release-schema.json) is incompatible because it nulls fields in `organizationReference`, which doesn't exist in 1.0

Conceptually, extensions (with the exception of the location extension), documentation about extensions and mechanisms to declare and validate extensions were only introduced in 1.1.

Apart from location, all the other extensions were developed against v1.1, so even where there isn't a technical incompatibility they could still be incompatible, e.g. a field in an extension referencing the `period` object could have a description which specifies use of the `period/durationInDays` or `period/maxExtentDate` fields, which were only introduced in 1.1.

For the above reasons and with a view to making data with extensions as usable as possible (e.g. by including the `extensions` key, introduced in 1.1, in the package metadata) and encouraging publishers to use the most recent version of the standard, I think it might best to keep the use of extensions restricted to 1.1.

1.1 is backward compatible, so if a publisher of 1.0 data wants to include extensions, updating the version number in the package metadata will only result in deprecating warnings rather than invalid data.

**Comment by duncandewhurst** [comment]
Also noting that the [qualification extension](https://github.com/open-contracting/ocds_qualification_extension/blob/master/README.md) is incompatible with 1.0, since it should only be used in the context of PPP profile, which is itself only compatible with 1.1.

**Comment by jpmckinney** [comment]
Noting that I should review this issue once the extensions explorer is further along, as it may no longer be relevant.

**Comment by jpmckinney** [comment]
Merged into https://github.com/open-contracting/extension-explorer/issues/3

**Comment by yolile** [comment]
I know that this is old, but the new field "schemas" is optional, but if you dont include it in your extension.json file the extension could not work. See this example for party/details: https://github.com/sdd1982/order this is a local extension that add the field "order" to the party details and doesnt include the "schema" field, if you test a release that use the extension you will notice that the extension is not properly included in the extended json schema, see here: http://standard.open-contracting.org/review/data/800a2cc0-6b38-4d40-a65a-c85bc02fdb4e

So maybe the "schemas" field should be required? Are the publishers that have local extension aware of this extension registry changes?

**Comment by jpmckinney** [comment]
It's optional because you can author an extension that only changes existing codelists. Please open an issue on CoVE: https://github.com/OpenDataServices/cove If the `schemas` field is missing, CoVE should attempt to request the `release-schema.json` file (to be backwards-compatible with old extensions, which didn't have a `schemas` field).
